File: Check if All Characters Have Equal Number of Occurrences/Check_if_All_Characters_Have_Equal_Number_of_Occurrences.cpp
class Solution {
public:
    bool areOccurrencesEqual(string s) {
        unordered_map<char,int> mpp;
        int maxi = 0;
        for(auto i: s) {
            mpp[i]++;
            maxi = max(maxi,mpp[i]);
        }
        for(auto i: mpp) {
            if(maxi != i.second) return false;
        }
        return true;
    }
};

File: Check if All Characters Have Equal Number of Occurrences/Check_if_All_Characters_Have_Equal_Number_of_Occurrences.java
class Solution {
    public boolean areOccurrencesEqual(String s) {
        Map<Character, Integer> mpp = new HashMap<>();
        int maxi = 0;
        for (char c : s.toCharArray()) {
            mpp.put(c, mpp.getOrDefault(c, 0) + 1);
            maxi = Math.max(maxi, mpp.get(c));
        }
        for (int count : mpp.values()) {
            if (maxi != count) {
                return false;
            }
        }
        return true;
    }
}